### DISCUSSION TITLE
Different TTL expiration per environment

### DISCUSSION THREAD

**Discussion by elmalakomar** [mia-platform/community#510] [Q&A] [answered]
### What product are you having troubles with?

Console

### What Console version are you using?

v12.0.2

### Description

It is possible in some way to define different TTL expire seconds based on the environment? 
Example is: I would like 7 days for the preproduction environment and 30 days for the production one

Thanks

**Comment by flower-of-the-bridges** [comment, marked as answer]
Hi @elmalakomar 

From the Console, it is only possible to specify a numeric value within the TTL form, so is not possible to use handlebar syntax.

However, keep in mind that TTL indexes should be used with collections having a small size of documents [due to their performances](https://www.mongodb.com/docs/v7.0/core/index-ttl/#timing-of-the-delete-operation):  when a collection has a great number of records, the deletion procedure triggered by the TTL expiration may happen during a peak of activity within your MongoDB cluster, slowing other operations happening.

Also, always from the MongoDB docs mentioned, TTL constraints are executed on the cluster every 60 seconds.

@albertotessarotto maybe we should consider adding a warning about TTL indexes on our docs.   

So, if your use case falls into this category, you may use a cronjob that performs the deletion periodically during hours when the database is not used by other processes (for example, at midnight). In that case, using the cronjob section of the Console, you can use handlebars to change scheduling time according to your environment.
